bot/cogs/define_cog.py

```diff
@@ -0,0 +1,130 @@
+#This contribution was made by: Rajat Sethi
+#Date: 12/15/2020
+
+import logging
+
+import discord
+import discord.ext.commands as commands
+from bot.messaging.events import Events
+from bot.bot_secrets import BotSecrets
+from bot.consts import Colors
+
+import json
+import aiohttp
+
+log = logging.getLogger(__name__)
+
+class defineCog(commands.Cog):
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.api_key = BotSecrets.get_instance().merriam_key
+        self.wordPages = []
+    
+    def getPageData(self, jsonData, word):
+        pages = []
+
+        #If the word is found, the JSON will return a dictionary of information. 
+        if (isinstance(jsonData[0], dict)):
+
+            #For words with several definitions, it will return several dictionaries. 
+            for wordData in jsonData:
+                #Stems of the given word (Past Tense, Future Tense, Perfect Tense, etc.)
+                wordStems = wordData.get('meta',{}).get('stems',[])
+                
+                #Syllables of the given word
+                syllableData = wordData.get('hwi',{}).get('hw',"")
+                
+                #Pronunciation of the given word (With those weird letters)
+                pronunc = []
+                prsData = wordData.get('hwi',{}).get('prs',[])
+                
+                for soundData in prsData:
+                    pronunc.append(soundData.get('mw',""))
+                
+                #Type of the given word (Noun, Verb, Adjective, etc.)
+                wordType = wordData.get('fl',"")
+                
+                #Definitions of the given word
+                definitions = []
+                defData = wordData.get('shortdef',[])
+                
+                for defin in defData:
+                    definitions.append(defin)
+                
+                #Turn data into one long string (represents a page)
+                template = "Tenses: "
+                for s in range(len(wordStems)):
+                    template += wordStems[s]
+                    if (s != len(wordStems) - 1):
+                        template += ", "
+                template += "\n"
+
+                template += "Syllables: " + syllableData + "\n"
```

Comment:
f strings always instead of concatenation, there is basically never a place in python where you should do str + str